### PR TITLE
Bug/gcom 1210 broken product image animation

### PR DESCRIPTION
**Pull request by Jessevdpoel**

### CODE REVIEW

File: packages/framer-scroller/components/ScrollerButton.tsx [hunk 1]
```diff
@@ -27,9 +27,17 @@ export const ScrollerButton = m(
       ...buttonProps
     } = props
 
+    const [isScrolling, setIsScrolling] = useState(false)
```
Comment (paales) [inline packages/framer-scroller/components/ScrollerButton.tsx:30]:
```suggestion
     const isScrolling = useRef(false)
```

File: packages/framer-scroller/components/ScrollerButton.tsx [hunk 2]
```diff
@@ -27,9 +27,17 @@ export const ScrollerButton = m(
...
+
     const { getSnapPosition, scroll } = useScrollerContext()
     const scrollTo = useScrollTo()
-    const handleClick = () => scrollTo(getSnapPosition(direction))
+    const handleClick = async () => {
+      if (!isScrolling) {
+        setIsScrolling(true)
+        await scrollTo(getSnapPosition(direction))
+        setIsScrolling(false)
+      }
```
Comment (paales) [inline packages/framer-scroller/components/ScrollerButton.tsx:39]:
```suggestion
      if (!isScrolling.current) {
        isScrolling.current = true
        await scrollTo(getSnapPosition(direction))
        isScrolling.current =  false
      }
```

**Review by paales — CHANGES_REQUESTED** [review]

File: packages/framer-scroller/components/ScrollerButton.tsx [hunk 3]
```diff
@@ -27,9 +27,16 @@ export const ScrollerButton = m(
       ...buttonProps
     } = props
 
+    const isScrolling = useRef(false)
     const { getSnapPosition, scroll } = useScrollerContext()
     const scrollTo = useScrollTo()
-    const handleClick = () => scrollTo(getSnapPosition(direction))
+    const handleClick = async () => {
+      if (!isScrolling.current) {
```
Comment (mikekeehnen) [inline packages/framer-scroller/components/ScrollerButton.tsx:34]: I think we already have a `scroll.animating`. You can use that variable to catch the "spammed" click

**Review by paales — CHANGES_REQUESTED** [review]
See changes from @mikekeehnen

File: packages/framer-scroller/components/ScrollerButton.tsx [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import { useMotionValueValue } from '@graphcommerce/framer-utils'
 import { Fab, FabProps, styled, SxProps, Theme } from '@mui/material'
 import { m, useTransform } from 'framer-motion'
-import React from 'react'
+import React, { useRef } from 'react'
```
Comment (paales) [inline packages/framer-scroller/components/ScrollerButton.tsx:4]: Not used

**Review by paales — APPROVED** [review]
This looks great! Last thing, can you clean up unused useRef?

Can we also create a followup ticket that will allow animating to the _next_ item when clicked multiple times?

**Pull request merged**